matrix/matrix.py

```diff
@@ -0,0 +1,161 @@
+        return cls(
+            tuple([tuple((0,) * i + (1,) + (0,) * (dimension - i - 1)) for i in
+                   range(dimension)]))
+
+    @classmethod
+    def ones(cls, dimension):
+        """Generate a matrix full of 1's."""
+        return cls(tuple(([tuple((1,) * dimension) for _ in
+                           range(dimension)])))
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}{self._matrix}'
+
+    def __str__(self):
+        """Return a string presentation of the matrix instance."""
+        return repr(self._matrix)
+
+    @property
+    def tuples(self):
+        """Return the matrix instance as a tuple of tuples."""
+        return self._matrix
+
+    def _matrix_by_matrix(self, other, operation):
+        """Calculate operation on two matrices.
+
+        Args:
+            operation (function): the function of the operator.
+            other (Matrix): the other matrix to operate on.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Boolean operations supports un-squared matrices but
+            mathematical operations don't.
+        """
+        if len(other.tuples) != len(self.tuples) and not any(
+                [operation == operator.eq, operation == operator.ne]):
+            raise ValueError("Matrices length don't match")
+
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(
+                    operation(self.tuples[i][j], other.tuples[i][j]))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(tuple(calculated_matrix))
+
+    def _by_scalar(self, scalar, operation):
+        """Calculate operation on matrix by scalar.
+
+        Args:
+            operation (function): the function of the operator.
+            scalar (number): the scalar to operate by.
+
+        Returns:
+            Matrix: the result of the operation.
+
+        Note:
+            Scalar must be a number. Operations with scalars are only * and /.
+        """
+        calculated_matrix = []
+
+        for i in range(len(self.tuples)):
+
+            calculated_line = []
+            for j in range(len(self.tuples)):
+                calculated_line.append(operation(self.tuples[i][j], scalar))
+
+            calculated_matrix.append(tuple(calculated_line))
+
+        return Matrix(tuple(calculated_matrix))
+
+    def __add__(self, other):
+        """Add two matrices."""
+        return self._matrix_by_matrix(other, operator.add)
+
+    def __sub__(self, other):
+        """Subtract two matrices."""
+        return self + (-1 * other)
+
+    def __mul__(self, other):
+        """Multiply two matrices or scalar by matrix."""
+        if isinstance(other, Matrix):
+            return self._matrix_by_matrix(other, operator.mul)
```

Comment:
It doesn't look like matrix multiplication
Are you sure about your implementation?

Reply:
What do you mean? It passed the tests.

Reply:
The tests also don't look right
Do you know how matrix multiplication works?

Reply:
Got the mistake, on it :)